### ISSUE TITLE
Consumer Group - Meaningful error on JoinGroupResponse error

### ISSUE THREAD

**Issue by dantswain** [kafkaex/kafka_ex#223]
Follow-up task to #219

**Comment by dantswain** [comment]
I realized I opened this without leaving much of a breadcrumb trail for what actually needs to be done.

https://github.com/kafkaex/kafka_ex/blob/ede2b09b6fd67decd1fea10aee0194a443fe5778/lib/kafka_ex/consumer_group/manager.ex#L203

```
    join_response = %JoinGroupResponse{error_code: :no_error} =
      KafkaEx.join_group(
        join_request,
        worker_name: worker_name,
        timeout: session_timeout + @session_timeout_padding
      )
```

This will result in a match fail if there is anything other than an "ok" response from the broker, which will crash the worker without much of a useful message to the user.  It should at least print out a more meaningful error message and we should evaluate if this really needs to crash the worker.

Additional comments from @joshuawscott https://github.com/kafkaex/kafka_ex/pull/219#discussion_r131994647

> This will crash if we get any error back from Kafka. I don't know if it's necessarily good to crash within a private function that's in an essentially private module. It will make it difficult to diagnose for a library user. Maybe we should crash with a helpful message here (like display a human-readable version of the error code that we got)

**Comment by dantswain** [comment]
Closed by #224
